trio/tests/test_threads.py

```diff
@@ -488,6 +497,18 @@ def thread_fn():
     await to_thread_run_sync(thread_fn)
     assert record == ["in thread", "back in trio"]
 
+    # Test correct error when passed sync function
+    def sync_fn():  # pragma: no cover
+        pass
+
+    def thread_fn():
+        from_thread_run(sync_fn)
+
+    with pytest.raises(TypeError) as excinfo:
+        await to_thread_run_sync(thread_fn)
```

Comment:
You don't need the separate thread_fn. You can say `await to_thread_run_sync(from_thread_run, sync_fn)`